Utils/src/LeptonProducer.cc

```diff
@@ -66,39 +66,85 @@ class LeptonProducer : public edm::global::EDProducer<> {
   edm::EDGetTokenT<edm::View<pat::MET>> metTok_;
   edm::EDGetTokenT<pat::PackedCandidateCollection> PFCandTok_;
   edm::EDGetTokenT<double> RhoTok_;
-  double minElecPt_, maxElecEta_, minMuPt_, maxMuEta_;
+  double minElecPt_, maxElecEta_, elecIsoValue_;
+  std::vector<double> eb_ieta_cut_, eb_deta_cut_, eb_dphi_cut_, eb_hovere_cut_, eb_ooeminusoop_cut_, eb_d0_cut_, eb_dz_cut_;
+  std::vector<int> eb_misshits_cut_;
+  std::vector<double> ee_ieta_cut_, ee_deta_cut_, ee_dphi_cut_, ee_hovere_cut_, ee_ooeminusoop_cut_, ee_d0_cut_, ee_dz_cut_;
+  std::vector<int> ee_misshits_cut_;
+  bool hovere_constant_;
+  double minMuPt_, maxMuEta_, muIsoValue_;
+  double muNormalizedChi2Max_, muChi2LocalPositionMax_, muTrkKink_, muValidFractionMin_;
+  std::vector<double> muSegmentCompatibilityMin_;
+  double mudBMax_, mudZMax_;
+  double tightMuNormalizedChi2Max_;
+  int tightMuNumberOfValidMuonHitsMin_, tightMuNumberOfMatchedStationsMin_;
+  double tightMudBMax_, tightMudZMax_;
+  int tightMuNumberOfValidPixelHitsMin_, tightMuTrackerLayersWithMeasurementMin_;
   bool useMiniIsolation_;
-  double muIsoValue_, elecIsoValue_;
+  std::vector<double> electronEAValues_, muonEAValues_;
   SUSYIsolation SUSYIsolationHelper;
 };
 
 //
 // constructors and destructor
 //
-LeptonProducer::LeptonProducer(const edm::ParameterSet& iConfig)
-{
+LeptonProducer::LeptonProducer(const edm::ParameterSet& iConfig):
   //register your products
-  MuonTag_                                 =         iConfig.getParameter<edm::InputTag >("MuonTag");
-  ElecTag_                                 =         iConfig.getParameter<edm::InputTag >("ElectronTag");
-  PrimVtxTag_=iConfig.getParameter<edm::InputTag>("PrimaryVertex");
-  minElecPt_=iConfig.getParameter<double>          ("minElecPt");
-  maxElecEta_=iConfig.getParameter<double>          ("maxElecEta");
-  minMuPt_=iConfig.getParameter<double>          ("minMuPt");
-  maxMuEta_=iConfig.getParameter<double>          ("maxMuEta");
-  useMiniIsolation_ = iConfig.getParameter<bool>("UseMiniIsolation");
-  muIsoValue_=iConfig.getParameter<double>          ("muIsoValue");
-  elecIsoValue_=iConfig.getParameter<double>          ("elecIsoValue");
-  metTag_   = iConfig.getParameter<edm::InputTag> ("METTag");
-  RhoTag_ = edm::InputTag("fixedGridRhoFastjetCentralNeutral");
-  PFCandTag_ = edm::InputTag("packedPFCandidates");
-  
-  MuonTok_ = consumes<edm::View<pat::Muon>>(MuonTag_);
-  ElecTok_ = consumes<edm::View<pat::Electron>>(ElecTag_);
-  PrimVtxTok_ = consumes<reco::VertexCollection>(PrimVtxTag_);
-  metTok_ = consumes<edm::View<pat::MET>>(metTag_);
-  PFCandTok_ = consumes<pat::PackedCandidateCollection>(PFCandTag_);
-  RhoTok_ = consumes<double>(RhoTag_);
-        
+  MuonTag_                               (iConfig.getParameter<edm::InputTag>("MuonTag")),
```

Comment:
this list starts to get a bit unwieldy... we might consider in the future making separate PSets for electrons and muons, just to organize it better

Reply:
I completely agree, but maybe we just do this for now and do that in a different PR?